src/instructlab/configuration.py

```diff
@@ -496,6 +496,7 @@ class _train(BaseModel):
     num_epochs: int
     effective_batch_size: int
     save_samples: int
+    checkpoint_at_epoch: bool = False
```

Comment:
how does this interact  w/ save-samples?

Reply:
they're mutually compatible- if checkpoint_at_epoch is set, we'll checkpoint at the end of epochs, separately from how frequently we checkpoint based on save_samples. There are different reasons for each.